e2-studio-star-rx72n-firmware/tests/test_rx_frame.c

```
@@ -2240,6 +2240,235 @@ void test_roundtrip_reset(void)
+ * @post bytes_discarded == 0 (function resets to 0 at entry; no sync = no advance)
+ *
+ * @note Not thread-safe; Unity runs tests sequentially on a single thread
+ *
+ * @see rx_frame_decode_with_resync() Function under test
+ * @see test_resync_dropped_byte_recovery() Complementary recovery-success test
+ *
+ * @since Version 1.1.0
+ */
+void test_resync_no_sync_found(void)
+{
+  uint8_t    buf[k_frame_min_size];
```

Clang (14.0.6)

[warning] variable 'buf' is not initialized

(cppcoreguidelines-init-variables)